resources/2024-06-resource-extensibility-feature-spec.md

```diff
@@ -0,0 +1,452 @@
+
+        Question :Can we build on top of `tsp init` which is used to set up a typespec project with a sample template? The other option is to use `rad resource-provider new` to scaffold a template typespec definition and the user takes care of the typespec dependencies and setup which might be a confusing.
+
+    1. Deb authors the schema for plaid in typespec format
+        
+        ```
+        import "@typespec/http";
+        import "@typespec/rest";
+        import "@typespec/openapi3";
+        import "@typespec/versioning";
+
+        using TypeSpec.Http;
+        using TypeSpec.Rest;
+        using TypeSpec.Versioning;
+
+        @service({title: "Contoso"})
+        @server("http://contoso.plaid.com", "endpoint for plaid service")
+        @versioned(Versions)
+        namespace Contoso;
+
+        @route("/planes/radius/Contoso.Messaging")
+        namespace plaid {
+            model plaid {
+                name:string;
+                description:string;
+                queue:string;
+            }
+        }
+
+        ```
+
+    1. VSCode Type spec extension warns Deb on the errors and warnings as he types
+     
+        ![alt text](2024-06-resource-extensbility/errors.png)
+
+    1. Deb uses intellisense to autocomplete the schema and properties
+
+        ![alt text](2024-06-resource-extensbility/intellisense.png)
+
+    1. Deb chooses a version for the schema
+
+        ```
+        import "@typespec/http";
+        import "@typespec/rest";
+        import "@typespec/openapi3";
+        import "@typespec/versioning";
+
+        using TypeSpec.Http;
+        using TypeSpec.Rest;
+        using TypeSpec.Versioning;
+
+        @service({title: "Contoso"})
+        @server("http://contoso.plaid.com", "endpoint for plaid service")
+        @versioned(Versions)
+        namespace Contoso;
+
+        @route("/planes/radius/Contoso.Messaging")
+        namespace plaid {
+            model plaid {
+                name:string;
+                description:string;
+                queue:string;
+            }
+        }
+
+        enum Versions {
+            v1,
+        }
+        ```
+   1. Deb registers the schema in Radius
+    
+        ```bash
+        rad resource-provider register contoso --template-kind typespec -e myenv -v v1
+        ```
+        Radius complies the schema and registers the custom resource type Plaid in UCP
+
+    1. Deb lists the resources types he has in his environment
+
+        ```bash
+        rad resource-provider list -e myenv
+        ```
+        Radius lists the custom resource types that Deb has registered in his environment
+
+2. Amy is a system integrator who is familiar with OpenAPI schema format and wants to author a custom resource PostgreSQL in Radius
+
+    1. Amy uses the radCLI to scaffold a yaml template definition
+
+        ```bash
+        rad resource-provider init plaid --template-kind yaml
```

Comment:
should this be:
 rad resource-provider init Applications --template-kind yaml
 since Applications is the rp used below?

Reply:
yes